### ISSUE TITLE
Issue with NGINX ingress pathType in apps 

### ISSUE THREAD

**Issue by cokiengchiara** [giantswarm/roadmap#325]
Mon, 10. May 2021 Update: I don't think my suggestions below works. I don't know enough about what caused this, where, and what can and should be done (if any) to prevent in the future. @ubergesundheit is collecting info and can decide further.

---

@giantswarm/team-halo , I got alerted that prometheus-operator upgrade failed. ([slack](https://gigantic.slack.com/archives/C03CPNRTS/p1619708710002800))

Was on a call as @pipo02mix worked w this together w @tomahawk28 and @webwurst 

- [ ] Consider modifying app charts to always include pathType, where apps require an ingress. Not sure about the implications. @giantswarm/app-squad-prometheus please think about it? Fer, Jihyuk, and Tobi can give more useful context. https://github.com/giantswarm/roadmap/issues/335
- [ ] From @pipo02mix : Another cool thing we could do is to see if the operator emits any metric when there are errors so we can catch it earlier

Don't think this is urgent. If it is, we'll get alerted with apps failing.

**Comment by cokiengchiara** [comment]
hi people, I attempted to modify the [ops recipe](https://intranet.giantswarm.io/docs/support-and-ops/ops-recipes/app-failed/) but realized I don't understand enough to make a good suggestion. 

Should it just be quick note in "Known Issues", or add a new section like below?

### Upgrade failed, cannot patch with kind Ingress

If you get an error regarding [Ingress path types](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types) like below

```status:
  appVersion: 0.47.0
  reason: 'Upgrade "prometheus" failed: cannot patch "prometheus-prometheus-oper-alertmanager"
    with kind Ingress: Ingress.extensions "prometheus-prometheus-oper-alertmanager"
    is invalid: spec.rules[0].http.paths[0].pathType: Required value: pathType must
    be specified && cannot patch "prometheus-prometheus-oper-prometheus" with kind
    Ingress: Ingress.extensions "prometheus-prometheus-oper-prometheus" is invalid:
    spec.rules[0].http.paths[0].pathType: Required value: pathType must be specified'```

Each path in an Ingress is required to have a corresponding path type. Modify the values file of the failed app with `pathType: ImplementationSpecific`

**Comment by pipo02mix** [comment]
AFAIU It is because ingress uses new apiGroup and they don't have a `pathType defined. To solve the most secure stuff is to use `pathType` with default value `ImplementationSpecific`. So yes for each path add the property `pathType` inside with default value

**Comment by cokiengchiara** [comment]
There was an ingress version without pathType. Now they created w a pathType.

What is the severity if it happens US time?

Collect info:
* Was it upgrade of cluster? Which version before? Which version upgraded to?

**Comment by pipo02mix** [comment]
> What is the severity if it happens US time?

I don't think it is high. It blocks a new upgrades or ingresses to be implemented but does not break existing ones.

> Was it upgrade of cluster? Which version before? Which version upgraded to?

They tried to upgrade the promehteus operator, as it was using a new `apiGroup` it triggers the `pathType` issue. 

Cluster `x6rtd   GK-software    gk-software-apps-germanywestcentral   14.0.2`
Prometheus operator `prometheus-operator-app-0.8.0` (upgraded from `prometheus-operator-app-0.7.1`)
Nginx controller `nginx-ingress-controller-app-1.15.0`

**Comment by ubergesundheit** [comment]
Released 0.8.1 of prometheus-operator-app (https://github.com/giantswarm/prometheus-operator-app/releases/tag/v0.8.1)

**Comment by cokiengchiara** [comment]
Closing in favor of https://github.com/giantswarm/roadmap/issues/335
